ddl/ddl_api.go

```diff
@@ -1940,3 +1990,44 @@ func validateCommentLength(vars *variable.SessionVars, comment string, maxLen in
 	}
 	return comment, nil
 }
+
+func buildPartitionInfo(meta *model.TableInfo, d *ddl, spec *ast.AlterTableSpec) (*model.PartitionInfo, error) {
+	part := &model.PartitionInfo{
+		Type:    meta.Partition.Type,
+		Expr:    meta.Partition.Expr,
+		Columns: meta.Partition.Columns,
+		Enable:  meta.Partition.Enable,
+	}
+	buf := new(bytes.Buffer)
+	for _, def := range spec.PartDefinitions {
+		for _, expr := range def.LessThan {
+			tp := expr.GetType().Tp
+			if !(tp == mysql.TypeLong || tp == mysql.TypeLonglong) {
+				expr.Format(buf)
+				if strings.EqualFold(buf.String(), "MAXVALUE") {
+					continue
+				}
+				buf.Reset()
+				return nil, infoschema.ErrColumnNotExists.GenByArgs(buf.String(), "partition function")
```

Comment:
Why return infoschema.ErrColumnNotExists here?

Reply:
It's consistent with mysql, for example.
```
mysql> ALTER TABLE table1 ADD PARTITION (
    -> PARTITION p3 VALUES LESS THAN (aa)
    -> );
ERROR 1054 (42S22): Unknown column 'aa' in 'partition function'
```